### ISSUE TITLE
Can addFunction support lambda?

### ISSUE THREAD

**Issue by wenyg** [vinniefalco/LuaBridge#245]
I want to pass some  localvariables from C++ to lua, but these variables can only be global variables.

This is my want
```C++
void my_process(){
  std::string my_data;
  luabridge::getGlobalNamespace(L)
      .beginNamespace ("ctx")
          .addFunction("set", [&my_data](){ return my_data.c_str();})
          .AddFunction("get", [&my_data](const char *data){ my_data = std::string(data);})
      .endNamespace();
  // process my_data in lua
  luaL_dofile(L, "./test.lua"); 
  // now, my_data is the processed data
}
```
`./test.lua` like this
```lua
data = ctx.get()
-- do something in lua
ctx.set(processed_data)
```
What should I do to pass C++ local variables to Lua, and then get it from lua?  Thank you very much

**Comment by kunitoki** [comment]
This is fully supported in https://github.com/kunitoki/LuaBridge3

**Comment by dmitry-t** [comment]
OK, once kunitoki's fork is fully compatible with the LuaBridge2, anybody can go to the LuaBridge3.
@kunitoki Can you confirm your version is fully compatible?

**Comment by kunitoki** [comment]
It's not fully compatible, i made some design decisions that were needed to support using the library without exceptions (games you know, game developers are allergic to exceptions), so some stuff has changed. There is a long list of breaking changes and removals. See

**Comment by dmitry-t** [comment]
I see. And I'd prefer C++17 :)

Most of code including older compiler support and e.g. `addData` are just for backward compatibility.

BTW the most unpleasant thing for one who wants to migrate from the LB2 to the LB3 is a different stack push signature. I guess this can be simplified by adding support for methods with the old signature using a meta-programming trick.
And you could also add the aliases like `addData` back and make them `[[deprecated]]`.
The only really breaking change then would be migration to C++17 compilers.

**Comment by kunitoki** [comment]
I want to polish the library, i'm seeking of removing legacy. If one wants addData can work with the older lb2.

**Comment by kunitoki** [comment]
Yeah for the stack changes, i might introduce more changes, like rolling an own expected as return value http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p0323r8.html
